agent/agent.go

```diff
@@ -67,28 +69,37 @@ type taskContext struct {
 
 // New creates a new Agent with some Options and a client.Communicator. Call the
 // Agent's Start method to begin listening for tasks to run.
-func New(opts Options, comm client.Communicator) *Agent {
+func New(opts Options, comm client.Communicator) (*Agent, error) {
 	comm.SetHostID(opts.HostID)
 	comm.SetHostSecret(opts.HostSecret)
 	agent := &Agent{
 		opts: opts,
 		comm: comm,
 	}
 
-	return agent
+	jpm, err := jasper.NewLocalManager(true)
+	if err != nil {
```

Comment:
Are you sure you want errors creating a process manager to prevent starting the agent?

Reply:
I think it's unavoidable.